scripts/Engineering_Diffraction_2.py

```diff
@@ -0,0 +1,26 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+# pylint: disable=invalid-name
+from __future__ import (absolute_import, division, print_function)
+from Engineering.gui.engineering_diffraction.engineering_diffraction import EngineeringDiffractionGui
+from qtpy import QtCore
+
+
```

Comment:
I'd glossed over this file a bit originally as I'd got used to seeing them. I can see how this is now confusing. How about we do something like:

```python
ENGINEERING_GUI = None

def guiInstance():
    global ENGINEERING_GUI
    if ENGINEERING_GUI is None:
        ENGINEERING_GUI = EngineeringDiffractionGui()
    return ENGINEERING_GUI

instance = guiInstance()
if not instance.isHidden():
    instance.setWindowState(instance.windowState() & ~QtCore.Qt.WindowMinimized
                                               | QtCore.Qt.WindowActive)
    instance.activateWindow()
else:
    instance.show()
```

Reply:
Won't `ENGINEERING_GUI` be set to None each time the script runs, so will re-create the GUI every time?

Reply:
Oh yes. I keep forgetting we run this script and not import the module.

Reply:
```
if 'ENGINEERING_GUI' not in globals():
    ENGINEERING_GUI = None
```
Maybe then just wrap that in an `if`? I think this is still a little clearer than before?